### PR TITLE
ci: integrate golangci-lint with new workflow

### PR DESCRIPTION
**Pull request by supitsdu**
### Tasks

- [x] Renamed .github/workflows/go.yml to go-standards.yml
- [x] Integrated golangci-lint into the Go Standards workflow
- [x] Updated golangci-lint configuration to optimize for CI workflows
- [x] Added comments and detailed explanations to the workflow and configuration files
- [x] Utilized latest GitHub actions versions, including actions/upload-artifact@v4

This commit aims to enhance code quality checks by incorporating comprehensive linting and ensuring compatibility with the latest GitHub actions.

### CODE REVIEW

File: .github/workflows/go-standards.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    # This job runs on the latest Ubuntu environment
+    runs-on: ubuntu-latest
+
+    steps:
+      # Step 1: Check out the repository
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Step 2: Set up Go environment
+      - name: Set up Go v1.22.3
```
Comment (ccoVeille) [inline .github/workflows/go-standards.yml:28]: I wouldn't use the version in the comment. The next time you will update the version node, you might forget to update this.

Comment (ccoVeille) [reply .github/workflows/go-standards.yml:28]: But, you can add a new steps and use a run cmd to launch `go version` so you have it in the GHA logs

File: .github/workflows/go-standards.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.22.3"
```
Comment (ccoVeille) [inline .github/workflows/go-standards.yml:31]: You could use 'latest' here if you prefer.

But I know it's open for debate

File: .github/workflows/go-standards.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+      # Step 3: Build the project
+      - name: Build the project
+        run: go build -v ./...
+        # Ensures build step is only attempted if the repository was checked out successfully
+        continue-on-error: false
```
Comment (ccoVeille) [inline .github/workflows/go-standards.yml:37]: This is the default value, so I don't think you need it

File: .github/workflows/go-standards.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+      # Step 4: Test the project
+      - name: Run tests
+        run: go test -v ./... -test.short
```
Comment (ccoVeille) [inline .github/workflows/go-standards.yml:41]:
```suggestion
        run: go test -v ./...
```

I'm unsure you should only run the short ones.

File: .golangci.yaml [hunk 5]
```diff
@@ -1,151 +1,82 @@
----
```
[the remark this answers is not in the document]
Comment (supitsdu) [reply .golangci.yaml:1]: Thanks! 

I wrote that by consulting the GolangCI docs and the helper linters command provided by the CLI tool. I didn't find a ready-made config that fit, so I used a trial-and-error approach, adjusting values, checking results, and researching as needed. It's definitely not the best option out there but it tackles the problem.

File: tests/tests.go [hunk 6]
```diff
@@ -1,51 +1,56 @@
...
+	_, err = file.WriteString(content)
+	if err != nil {
+		return nil, err
+	}
+
+	return file, nil
+}
+
+// IsCIEnvironment checks if the code is running in a CI environment.
+func IsCIEnvironment() bool {
```
[the remark this answers is not in the document]
Comment (supitsdu) [reply tests/tests.go:54]: Thanks, I presumed I would use this quite a few later, always nice to have a helper/utility function and avoid code repetitions.

File: tests/tests.go [hunk 7]
```diff
@@ -1,51 +1,56 @@
...
+	return os.Getenv("GITHUB_ACTIONS") == "true"
```
[the remark this answers is not in the document]
Comment (supitsdu) [reply tests/tests.go:55]: Oh yes, I've searched for a way to use environment variables approach as you suggested before and found this at [Learn Github Actions - Variables](https://docs.github.com/en/actions/learn-github-actions/variables)

It says the following:
`GITHUB_ACTIONS`
Always set to `true` when GitHub Actions is running the workflow. You can use this variable to differentiate when tests are being run locally or by GitHub Actions.

Which as the descriptions suggests it fits perfectly for this case.

**Pull request merged**